payu/models/mitgcm.py

```diff
@@ -37,9 +37,23 @@ def __init__(self, expt, name, config):
     def setup(self):
 
         # TODO: Find a better place to generate this list
-        self.config_files = [f for f in os.listdir(self.control_path)
+        files = [f for f in os.listdir(self.control_path)
                              if f.startswith('data')]
-        self.config_files.append('eedata')
+        files.append('eedata')
+
+        # Rudimentary check that matching files are namelists. Can only check if 
+        # namelist is empty. May excluded false positives, but these are # devoid 
+        # of useful information in that case
+        for fname in files:
+            nml_parser = f90nml.Parser()
+            nml_parser.comment_tokens += '#'
+
+            data_nml = nml_parser.read(fname)
+            if len(data_nml) > 0:
+                self.config_files.append(fname)
+            else:
+                print("Excluding {0} from list of configuration files: "\
+                      "assumed to be not a namelist file (or empty)".format(fname))
```

Comment:
can probably ditch the slash, I think string concatenation is automatic inside of parentheses.

Reply:
Fixed in the PEP8 cleanup